src/Components/FitnessFilters.js

```diff
@@ -0,0 +1,162 @@
+import filterStyle from '../styles/filters.style';
+import { Levels, Intensities, EquipmentLabels, SeasonTimings } from '../Fixtures';
+import Button from './shared/FilterButton';
+import HeaderButton from './shared/HeaderButton';
+
+class FitnessFilters extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedLevel: undefined,
+      selectedIntensity: undefined,
+      selectedEquipmentLabel: undefined,
+      selectedSeasonTiming: undefined,
+      selectedGoal: undefined,
+      durationInMinutes: undefined,
+      displayedDrills: this.props.route.params.initialData,
+    };
+
+    this.onDurationInMinutesChange = this.onInputChange.bind(this, 'durationInMinutes');
+  }
+
+  onPressedChange(target, value) {
+    this.setState(prevState => {
+      const newValue = value === prevState[target] ? undefined : value;
+      return {
+        [target]: newValue,
+      };
+    }, this.applyFilters);
+  }
+
+  onInputChange(target, value) {
+    this.setState({ [target]: value }, this.applyFilters);
+  }
+
+  applyFilters() {
+    const {
+      selectedLevel,
+      selectedIntensity,
+      selectedEquipmentLabel,
+      selectedSeasonTiming,
+      selectedGoal,
+      durationInMinutes,
+    } = this.state;
+    let newData = this.props.route.params.initialData;
+
+    if (selectedLevel) newData = newData.filter(drill => drill.level === selectedLevel);
+    if (selectedIntensity) newData = newData.filter(drill => drill.intensity === selectedIntensity);
+    if (selectedEquipmentLabel) newData = newData.filter(drill => drill.equipmentLabel === selectedEquipmentLabel);
+    if (selectedSeasonTiming) newData = newData.filter(drill => drill.seasonTiming === selectedSeasonTiming);
+    if (selectedGoal) newData = newData.filter(drill => drill.goals.includes(selectedGoal));
+    if (durationInMinutes) newData = newData.filter(drill => drill.durationInMinutes <= durationInMinutes);
+
+    this.setState({ displayedDrills: newData });
+  }
+
+  componentDidMount() {
+    this.props.navigation.setOptions({
+      headerRight: () => (
+        <HeaderButton
+          onPress={() => {
+            this.props.navigation.navigate(this.props.route.params.previousScreen, {
+              filteredDrills: this.state.displayedDrills,
+              type: this.props.route.params.previousType,
+            });
+          }}
+        />
+      ),
+    });
+  }
+
+  render() {
+    const {
+      selectedLevel,
+      selectedIntensity,
+      selectedEquipmentLabel,
+      selectedSeasonTiming,
+      selectedGoal,
+      durationInMinutes,
+    } = this.state;
+    return (
+      <View style={filterStyle.wrapper}>
+        <Text style={filterStyle.counter}>{this.state.displayedDrills.length} drills available</Text>
+        <ScrollView contentContainerStyle={filterStyle.filters}>
+          <Text style={filterStyle.filterTitle}>Level</Text>
```

Comment:
I'm thinking about extracting a FilterComponent (or ButtonFilter and SliderFilter) to DRY this a bit